library/src/domain/Transport.ts

```diff
@@ -0,0 +1,38 @@
+import { pipe } from "fp-ts/function";
+import * as Codec from "io-ts/Codec";
+import { Co2Eq } from "./Co2Eq";
+import { fromClassCodec } from "./io-ts";
+
+export class Transport {
+  type;
+  co2Eq;
+  explanation;
```

Comment:
`TransportMode` ? `TransportType` ? GTFS uses `RouteType` for this, but we don't have Routes

Reply:
Actually Transport may also be a combination of multiple means. For example, on the page for the search "Paris - Ajaccio", we should display two alternatives:
- the plane option
- the train + boat option

So the current structure may not be enough, new proposal:
- `Trip` still represents the wish of the user: traveling from `origin` to `destination`.
- `Trip.routes: Route[]`, a `Route` would represent a solution to go from `origin` to `destination`. It has a total co2eq. It can be composed of multiple means of transportation, so:
- `Route.segments: Segment[]`, to represent the details of a route, a `Segment` has its own `origin & destination`, a `type: flight | train | boat`, also a co2eq (`sum(route.segments, segment => segment.co2eq) === route.co2eq`), why not a distance

To simplify the front, we could add a `type` to `Route` which could be `'flight' | 'train' | 'boat' | 'train+boat'` depending on the segments.